### PR TITLE
Allow duplicate relations for JOIN

### PR DESCRIPTION
**Pull request by harunurhan**
Addresses #4775 and its comments.

Duplicate relations are allowed, as long as common columns are not referenced. 

After removing the duplicate check now, if common columns of duplicate relations are referenced it gives error with `column <first-referenced-common-column-name> is ambiguous`

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java [hunk 1]
```diff
@@ -106,8 +106,7 @@
     public void testDuplicateRelation()
             throws Exception
     {
-        assertFails(DUPLICATE_RELATION, "SELECT * FROM t1 JOIN t1 USING (a)");
```
Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: You should add tests to AbstractTestQueries to ensure this works properly now

File: presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java [hunk 2]
```diff
@@ -106,8 +106,7 @@
...
-        assertFails(DUPLICATE_RELATION, "SELECT * FROM t1 x JOIN t2 x USING (a)");
+        assertFails(DUPLICATE_RELATION, "WITH x as (SELECT 1), x as (SELECT 2) SELECT * from x");
```
Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: This test seems unrelated. 

Also, I'm not even sure that's invalid, but I need to check the SQL spec. It's possible that the second `x` hides the first one.

Comment (harunurhan) [inline presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java]: I'll remove it and testDuplicateRelation() since it will become empty.

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java [hunk 3]
```diff
@@ -2361,6 +2361,14 @@ public void testRightJoinWithNullValues()
     }
 
     @Test
+    public void testJoinWithDuplicateRelations()
+            throws Exception
+    {
+        assertQuery("SELECT * FROM orders JOIN orders USING (orderkey)");
+        assertQuery("SELECT * FROM lineitem x JOIN orders x USING (orderkey)");
```
Comment (harunurhan) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: @martint 
Although I am able run these queries on Presto, these tests cause error in h2 parser.
```
testJoinWithDuplicateRelations(com.facebook.presto.tests.TestLocalQueries)  Time elapsed: 0.708 sec  <<< FAILURE!
org.skife.jdbi.v2.exceptions.UnableToCreateStatementException: org.h2.jdbc.JdbcSQLException: Syntax error in SQL statement "SELECT * FROM ORDERS JOIN ORDERS USING ([*]ORDERKEY) "; SQL statement:
SELECT * FROM orders JOIN orders USING (orderkey) [42000-189] [statement:"SELECT * FROM orders JOIN orders USING (orderkey)", located:"SELECT * FROM orders JOIN orders USING (orderkey)", rewritten:"SELECT * FROM orders JOIN orders USING (orderkey)", arguments:{ positional:{}, named:{}, finder:[]}]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.message.DbException.getSyntaxError(DbException.java:191)
	at org.h2.command.Parser.getSyntaxError(Parser.java:517)
	at org.h2.command.Parser.prepareCommand(Parser.java:244)
	at org.h2.engine.Session.prepareLocal(Session.java:461)
	at org.h2.engine.Session.prepareCommand(Session.java:403)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1188)
	at org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:72)
	at org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:276)
	at org.skife.jdbi.v2.DefaultStatementBuilder.create(DefaultStatementBuilder.java:52)
	at org.skife.jdbi.v2.SQLStatement.internalExecute(SQLStatement.java:1304)
	at org.skife.jdbi.v2.Query.fold(Query.java:173)
	at org.skife.jdbi.v2.Query.list(Query.java:82)
	at org.skife.jdbi.v2.Query.list(Query.java:75)
	at com.facebook.presto.tests.H2QueryRunner.execute(H2QueryRunner.java:125)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:85)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:94)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:88)
	at com.facebook.presto.tests.AbstractTestQueries.testJoinWithDuplicateRelations(AbstractTestQueries.java:2367)
...
```

Comment (harunurhan) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java]: Nevermind, I realized that `assertQuery` also takes `expectedSql` which is executed by h2 runner. So, I added equivalent queries that are also valid for h2.

**Comment by harunurhan** [comment]
@martint addressed your comments.

**Comment by martint** [comment]
Merged, thanks!
